x/programs/rust/examples/multisig/src/lib.rs

```diff
@@ -0,0 +1,392 @@
+    yea: u32,
+    nay: u32,
+    executed: bool,
+    voters: HashSet<Address>,
+}
+
+#[derive(Debug, BorshSerialize, BorshDeserialize)]
+pub enum ProposalError {
+    AlreadyExecuted,
+    AlreadyVoted,
+    InexistentProposal,
+    QuorumNotReached,
+    ExecutionFailed(ExternalCallError),
+}
+
+#[state_keys]
+pub enum StateKeys {
+    LastProposalId,
+    Proposals(u32),
+}
+
+#[public]
+pub fn propose(
+    context: Context<StateKeys>,
+    to: Program,
+    method: String,
+    data: Vec<u8>,
+    value: u64,
+) -> u32 {
+    let program = context.program();
+    let last_id = proposal_id(program);
+
+    program
+        .state()
+        .store_by_key(
+            StateKeys::Proposals(last_id),
+            &Proposal {
+                to,
+                method,
+                data,
+                value,
+                yea: 1,
+                nay: 0,
+                executed: false,
+                voters: HashSet::from([context.actor()]),
+            },
+        )
+        .expect("state corrupt");
+
+    program
+        .state()
+        .store_by_key(StateKeys::LastProposalId, &(last_id + 1))
+        .unwrap();
+
+    last_id
+}
+
+#[public]
+pub fn vote(context: Context<StateKeys>, id: u32, yea: bool) -> Result<(), ProposalError> {
+    let program = context.program();
+    let mut proposal = proposal_at(program, id).ok_or(ProposalError::InexistentProposal)?;
+
+    let actor = context.actor();
+    if proposal.voters.contains(&actor) {
+        return Err(ProposalError::AlreadyVoted);
+    }
+    if proposal.executed {
+        return Err(ProposalError::AlreadyExecuted);
+    }
+
+    if yea {
+        proposal.yea += 1;
+    } else {
+        proposal.nay += 1;
+    }
+    proposal.voters.insert(actor);
+
+    program
+        .state()
+        .store_by_key(StateKeys::Proposals(id), &proposal)
+        .expect("state corrupt");
+
+    Ok(())
+}
+
+#[public]
+pub fn execute(
+    context: Context<StateKeys>,
+    proposal_id: u32,
+    max_units: u64,
+) -> Result<DeferDeserialize, ProposalError> {
+    let program = context.program();
+    let mut proposal =
+        proposal_at(program, proposal_id).ok_or(ProposalError::InexistentProposal)?;
+
+    if proposal.executed {
+        return Err(ProposalError::AlreadyExecuted);
+    }
+    if !quorum_reached(&proposal) {
+        return Err(ProposalError::QuorumNotReached);
+    }
+
+    proposal.executed = true;
+
+    proposal
+        .to
+        .call_function::<DeferDeserialize>(
+            &proposal.method,
+            &proposal.data,
+            max_units,
+            proposal.value,
+        )
+        .map_err(ProposalError::ExecutionFailed)
+}
+
+#[public]
+pub fn pub_proposal_at(context: Context<StateKeys>, proposal_id: u32) -> Option<Proposal> {
+    proposal_at(context.program(), proposal_id)
+}
+
+#[public]
+pub fn pub_proposal_id(context: Context<StateKeys>) -> u32 {
+    proposal_id(context.program())
+}
+
+#[public]
+pub fn pub_quorum_reached(
```

Comment:
```suggestion
pub fn proposal_at(context: Context<StateKeys>, proposal_id: u32) -> Option<Proposal> {
    proposal_at(context.program(), proposal_id)
}

#[public]
pub fn proposal_id(context: Context<StateKeys>) -> u32 {
    proposal_id(context.program())
}

#[public]
pub fn quorum_reached(
```

Reply:
That won't work because these function names are conflicting with the non-public functions.